packages/aragon-api/src/index.js

```diff
@@ -228,7 +228,9 @@ export class AppProxy {
     // (because of the `Promise.resolve`).
     const wrappedReducer = (state, event) =>
       from(
-        Promise.resolve(reducer(state, event))
+        Promise.resolve(reducer(state, event)).catch(e => console.error(
+          `Error from app reducer:`, e)
```

Comment:
I'm wondering what the correct behaviour for us should be. Should we re-throw the error, continue like nothing happened (as we do here, but return `undefined` for the new state), or try to give the error back to the frontend.

I'd recommend at least rethrowing the error for now, so the observable stream stops where it is, but we should also think about having a mechanism for frontends to detect their reducer has failed.

Reply:
Good point. I didn't think that far as I was a little averse to API changes and wanted to get logging before improving the mechanism. Nevertheless, this is worth looking deeper into. 

I'll do some more testing and report back.